src/OpenLoco/src/Drawing/SoftwareDrawingContext.cpp

```diff
@@ -1509,16 +1510,124 @@ namespace OpenLoco::Drawing
+
+            if (!startRect.intersects(renderTargetRect))
+            {
+                return;
+            }
+
+            auto leftX = left - rt.x;
+            if (leftX < 0)
+            {
+                leftX = 0;
+            }
+
+            auto rightX = right - rt.x + 1;
+            if (rightX > rt.width)
+            {
+                rightX = rt.width;
+            }
+
+            auto topY = top - rt.y;
+            if (topY < 0)
+            {
+                topY = 0;
+            }
+
+            auto bottomY = bottom - rt.y + 1;
+            if (bottomY > rt.height)
+            {
+                bottomY = rt.height;
+            }
+
+            auto drawRect = Rect::fromLTRB(leftX, topY, rightX, bottomY);
+
+            if (flags == RectFlags::none) // Regular fill
+            {
+                auto* dst = drawRect.top() * (rt.width + rt.pitch) + drawRect.left() + rt.bits;
+                const auto step = rt.width + rt.pitch;
+
+                for (auto y = 0; y < drawRect.height(); y++)
+                {
+                    std::fill_n(dst, drawRect.width(), colour);
+                    dst += step;
+                }
+            }
+            else if ((flags & RectFlags::transparent) != RectFlags::none)
+            {
+                auto* dst = rt.bits
+                    + static_cast<uint32_t>((drawRect.top() >> rt.zoomLevel) * ((rt.width >> rt.zoomLevel) + rt.pitch) + (drawRect.left() >> rt.zoomLevel));
+
+                auto paletteMap = Gfx::PaletteMap::getForColour(static_cast<ExtColour>(colour));
+                if (paletteMap.has_value())
+                {
+                    const auto& paletteEntries = paletteMap.value();
+                    const auto scaledWidth = drawRect.width() >> rt.zoomLevel;
+                    const auto scaledHeight = drawRect.height() >> rt.zoomLevel;
+                    const auto step = (rt.width >> rt.zoomLevel) + rt.pitch;
+
+                    // Fill the rectangle with the colours from the colour table
+                    for (auto y = 0; y < scaledHeight; y++)
+                    {
+                        auto* nextDst = dst + step * y;
+                        for (auto x = 0; x < scaledWidth; x++)
+                        {
+                            auto index = *(nextDst + x);
+                            *(nextDst + x) = paletteEntries[index];
+                        }
+                    }
+                }
+            }
+            else if ((flags & RectFlags::crossHatching) != RectFlags::none)
+            {
+                auto* dst = (drawRect.top() * (rt.width + rt.pitch)) + drawRect.left() + rt.bits;
+                const auto step = rt.width + rt.pitch;
+
+                for (auto y = 0; y < drawRect.height(); y++)
+                {
+                    auto* nextDst = dst + step * y;
+                    bool fillPixel = y % 2; // alternate filling first pixel for even and odd lines
```

Comment:
I'm sure all compilers will optimise this into a & but just incase can you make this y & 1. I need to check if this makes sense I've a feeling it needs to be related to rt.x, y to render correctly.

Reply:
Just checked and yes it does need the old code to render correctly unfortunately.